src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -522,12 +522,16 @@ public function dumpFile($filename, $content)
             throw new IOException(sprintf('Unable to write to the "%s" directory.', $dir), 0, null, $dir);
         }
 
+        // Will create a temp file with 0600 access rights
+        // when the filesystem supports chmod.
         $tmpFile = $this->tempnam($dir, basename($filename));
 
         if (false === @file_put_contents($tmpFile, $content)) {
             throw new IOException(sprintf('Failed to write file "%s".', $filename), 0, null, $filename);
         }
 
+        // Ignore for filesystems that do not support umask
+        @chmod($tmpFile, 0666 & ~umask());
```

Comment:
@xabbuh, I agree you could see it that way, lets not take the umask into account and create a separate pull request for it.